src/api.jl

```diff
@@ -4338,8 +4364,18 @@ XPRSchgrowtype,
 XPRSgetrowrange,
 XPRSgetrows.
 """
-function getrowtype(prob::XpressProblem, _srowtype, first::Int, last::Int)
-    @checked Lib.XPRSgetrowtype(prob, _srowtype, first, last)
+function getrowtype(prob::XpressProblem, first::Integer, last::Integer)
+    n_elems = last - first + 1
+    if n_elems <= 0
+        n_elems = n_constraints(prob)
+        first = 0
+        last = n_elems - 1
+    else
+        first = first - 1
+    end
```

Comment:
this coudl be a function:
`n_elems, first, last = myfunc(first, last)`

Reply:
Yes, agreed. I'll make the change. It would need to take a function argument too, since it can `get_marks(prob, first, last, n_constraints)` or `get_marks(prob, first, last, n_rows)`.